compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangTupleTypeNode.java

```diff
@@ -37,11 +37,11 @@
 public class BLangTupleTypeNode extends BLangType implements TupleTypeNode {
 
     // BLangNodes
-    public List<BLangType> memberTypeNodes = new ArrayList<>();
+    public List<BLangMemberTypeNode> memberTypeNodes = new ArrayList<>();
```

Comment:
Did we consider maintaining this as a list of `BLangSimpleVariable` similar to how we do with record/object fields? Asking because the parallel is record/object field, rather than a new type node.

Reply:
That was not considered earlier since the tuple members are not variables, but since now we're using `BVarSymbol` in `BTupleType` anyway maybe we can do similarly.

Reply:
Yes, since we need the symbols and are modeling it similar to records, makes sense to follow the same for tuples.